include/valijson/validation_results.hpp

```diff
@@ -51,6 +51,36 @@ class ValidationResults
     };
 
     /**
+     * @brief  Iterator for all validation errors.
+     *
+     * This struct is used to get a reference to an iterator for errors.
+     */
+    struct ErrorRange
+    {
+        typedef std::deque<Error>::const_iterator ErrorIterator;
+        ErrorRange(const std::deque<Error> &errors) : m_errors(errors) {};
+
+        ErrorIterator begin() {
+            return m_errors.begin();
+        }
+
+        ErrorIterator end() {
+            return m_errors.end();
+        }
+
+    private:
+        const std::deque<Error> &m_errors;
```

Comment:
This is really the only part of this PR that concerns me. I think it would be better for the ErrorRange struct to take a complete copy of the std::deque. This would ensure that calling pushError or popError on the original ValidationResults instance does not invalidate the iterators returned by begin() and end().

Unfortunately, the pushError() and popError() functions were one of the more haphazardly designed parts of Valijson, and I didn't really think about use cases beyond some test code that I was writing at the time. But now it's more or less part of the API.